flow/project.py

```diff
@@ -3796,14 +4232,22 @@ def _select_jobs_from_args(self, args):
                 "Cannot provide both -j/--job-id and -f/--filter or --doc-filter in combination.")
 
         if args.job_id:
-            try:
-                return [self.open_job(id=job_id) for job_id in args.job_id]
-            except KeyError as error:
-                raise LookupError("Did not find job with id {}.".format(error))
-        else:
+            aggregate = set()  # set in order to avoid duplicate ids
+            for id in args.job_id:
+                if id.startswith('agg'):
+                    aggregate.add(self._get_aggregate_from_id(id))
+                    continue
```

Comment:
Rather than a `continue` I would like to see an `else` block. Also, this code is used a few times (such as `_main_exec`). It may be worth refactoring into a separate function. (I am referring to the code to get aggregates from ids)

Reply:
Noted, will be tackled in a later PR.